File: src/api/group.ts
// group 命令薄包装。
// 对齐 src-tauri/src/commands/group.rs：list_groups / create_group / update_group / delete_group。

import type { Group, GroupInput, GroupUpdate } from '@/types/group'
import { invokeCmd } from './invoke'

/** 列出全部分组（按 order 升序） */
export function listGroups(): Promise<Group[]> {
  return invokeCmd<Group[]>('list_groups')
}

/** 新建分组 */
export function createGroup(input: GroupInput): Promise<Group> {
  return invokeCmd<Group>('create_group', { input })
}

/** 更新分组（name / order 可选） */
export function updateGroup(id: number, input: GroupUpdate): Promise<Group> {
  return invokeCmd<Group>('update_group', { id, input })
}

/** 删除分组（其下项目 group_id 置空） */
export function deleteGroup(id: number): Promise<void> {
  return invokeCmd<void>('delete_group', { id })
}

File: src-tauri/src/commands/group.rs
//! group 命令薄层
//!
//! 签名对齐 docs/03-命令清单.md。从 AppHandle 取 pool，转调 GroupService。

use tauri::{AppHandle, Runtime};

use crate::db;
use crate::error::AppResult;
use crate::models::{Group, GroupInput, GroupUpdate};
use crate::services::GroupService;

#[tauri::command]
pub async fn list_groups<R: Runtime>(app: AppHandle<R>) -> AppResult<Vec<Group>> {
    let pool = db::pool(&app)?;
    GroupService::list(&pool).await
}

#[tauri::command]
pub async fn create_group<R: Runtime>(
    app: AppHandle<R>,
    input: GroupInput,
) -> AppResult<Group> {
    let pool = db::pool(&app)?;
    GroupService::create(&pool, input).await
}

#[tauri::command]
pub async fn update_group<R: Runtime>(
    app: AppHandle<R>,
    id: i64,
    input: GroupUpdate,
) -> AppResult<Group> {
    let pool = db::pool(&app)?;
    GroupService::update(&pool, id, input).await
}

#[tauri::command]
pub async fn delete_group<R: Runtime>(app: AppHandle<R>, id: i64) -> AppResult<()> {
    let pool = db::pool(&app)?;
    GroupService::delete(&pool, id).await
}
